src/table.rs

```diff
@@ -1914,6 +1931,23 @@ impl<T> ExactSizeIterator for IterMut<'_, T> {
 
 impl<T> FusedIterator for IterMut<'_, T> {}
 
+// FIXME(#26925) Remove in favor of `#[derive(Clone)]`
+impl<'a, T> Clone for IterMut<'a, T> {
```

Comment:
Cloning a mutable iterator is unsound since it could be used to create multiple aliasing mutable references. All the other impls seem fine though.

Reply:
Sorry, I'm still new to Rust. I understand your point and it makes sense, so I removed the implementation of Clone here. And in fact, looking at my own code, my wrapping iterators haven't been implementing this either.

However, if this leads to unsoundness, how come I was able to implement the logic in safe code and the compiler didn't complain? Is something in HashTable incorrectly breaking the rules somehow which allowed my transgression?

Reply:
This isn't entirely safe: if you look above, the `Iterator` trait implementation for `IterMut` uses unsafe code. That unsafe code is only valid if it is the only instance of `IterMut` for a table.

Reply:
This isn't resolved, it seem you have re-added the incorrect `Clone` impl.

Reply:
OK, redeleted. Sorry about that.

Reply:
On the other hand, `Debug for IterMut` is fine.

Reply:
To implement Debug, I needed Clone, so I just punted on IterMut alltogether.

Reply:
The map `IterMut` does it through a private `iter()` that only clones the `inner: RawIter`.

https://github.com/rust-lang/hashbrown/blob/a69af93fe1aca2b2eedf1e3e01ef7eaef10be927/src/map.rs#L3175-L3183

https://github.com/rust-lang/hashbrown/blob/a69af93fe1aca2b2eedf1e3e01ef7eaef10be927/src/map.rs#L2137-L2146